tests/Functional/ViewHelpers/ConstantViewHelperTest.php

```diff
@@ -0,0 +1,92 @@
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+namespace TYPO3Fluid\Fluid\Tests\Functional\ViewHelpers;
+
+use TYPO3Fluid\Fluid\Tests\Functional\AbstractFunctionalTestCase;
+use TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\BackedEnumExample;
+use TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\ClassConstantsExample;
+use TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\EnumExample;
+use TYPO3Fluid\Fluid\View\TemplateView;
+
+final class ConstantViewHelperTest extends AbstractFunctionalTestCase
+{
+    /**
+     * @test
+     */
+    public function renderThrowsExceptionOnNonStringValue(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $value = new \stdClass();
+        $view = new TemplateView();
+        $view->assignMultiple(['value' => $value]);
+        $view->getRenderingContext()->setCache(self::$cache);
+        $view->getRenderingContext()->getTemplatePaths()->setTemplateSource('<f:constant value="{value}" />');
+        $view->render();
+    }
+
+    /**
+     * @test
+     */
+    public function renderThrowsErrorOnUndefinedConstant(): void
+    {
+        $this->expectException(\Error::class);
+        $value = 'FOO';
+        $view = new TemplateView();
+        $view->assignMultiple(['value' => $value]);
+        $view->getRenderingContext()->setCache(self::$cache);
+        $view->getRenderingContext()->getTemplatePaths()->setTemplateSource('<f:constant value="{value}" />');
+        $view->render();
+    }
+
+    public static function renderDataProvider(): \Generator
+    {
+        yield 'Value is built-in PHP constant' => [
+            'PHP_INT_MAX',
+            PHP_INT_MAX,
+        ];
+
+        yield 'Value is class constant' => [
+            '\TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\ClassConstantsExample::FOO',
+            ClassConstantsExample::FOO,
+        ];
+
+        yield 'Value is backed enum case' => [
+            '\TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\BackedEnumExample::BAR',
+            BackedEnumExample::BAR,
+        ];
+
+        yield 'Value is enum case' => [
+            '\TYPO3Fluid\Fluid\Tests\Functional\Fixtures\Various\EnumExample::FOO',
+            EnumExample::FOO,
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider renderDataProvider
+     */
+    public function render(mixed $value, mixed $expected): void
+    {
+        $view = new TemplateView();
+        $view->assignMultiple(['value' => $value]);
+        $view->getRenderingContext()->setCache(self::$cache);
+
+        $templateSources = [
+            '<f:constant value="{value}" />',
+            '<f:constant>{value}</f:constant>',
+            '{f:constant(value: \'{value}\')}',
+        ];
+
+        foreach ($templateSources as $templateSource) {
+            $view->getRenderingContext()->getTemplatePaths()->setTemplateSource($templateSource);
+            self::assertSame($expected, $view->render());
```

Comment:
Currently, we execute all functional test cases twice to test both uncached and cached behavior, see other VH tests. Can you add this to your test as well?